File: pset1/cash/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    float dolars;
    do
    {

        dolars = get_float("change owed:");


    }
    while (dolars < 0);



    int cents = round(dolars * 100);
    int coins;

    while (cents > 25)
    {

        coins = cents / 25;
        cents = cents % 25;

    }
    while (cents > 10 && cents < 25)
    {

        coins += cents / 10;
        cents = cents % 10;

    }
    while (cents > 5 && cents < 10)
    {

        coins += cents / 5;
        cents = cents % 5;

    }
    while (cents > 0 && cents < 5)
    {

        coins += cents / 1;
        cents = cents % 1;

    }
    printf("coins:%i \n", coins);
}

File: pset6/cash/cash.py
import re
from math import floor
while True:

    dolars = input("change owed:")
    x = re.search("\d\.*", dolars)
    y = re.search("\D", dolars)
    if x:
        while float(dolars) < 0:
            print("you should enter valid number:")
            dolars = input("change owed:")
        cents = floor(float(dolars) * 100)
        break;

    elif y:
        print("please enter a valid input:")


quarters = cents // 25

dimes = (cents % 25) // 10

nickels = ((cents % 25) % 10) // 5

pennies = ((cents % 25) % 10) % 5


print(f"coins:{quarters + dimes + nickels + pennies}")